backend/infrahub/core/query/relationship.py

```diff
@@ -948,3 +957,162 @@ async def get_count_per_peer(self) -> dict[str, int]:
+                }
+
+                query += sub_query
+
+        # We only want to return uuid/kind of `Node` connected through `IS_RELATED` edges.
+        query += """
+        CALL {
+            WITH rl
+            MATCH (rl)-[active_edge:IS_RELATED]->(n)
+            WHERE %(active_rel_filter)s AND active_edge.status ="active"
+            CREATE (rl)-[deleted_edge:IS_RELATED $rel_prop]->(n)
+            SET deleted_edge.hierarchy = active_edge.hierarchy
+            WITH rl, active_edge, n
+            WHERE active_edge.branch = $branch AND active_edge.to IS NULL
+            SET active_edge.to = $at
+            RETURN
+                n.uuid as uuid,
+                n.kind as kind,
+                rl.name as rel_identifier,
+                "outbound" as rel_direction
+
+            UNION
+
+            WITH rl
+            MATCH (rl)<-[active_edge:IS_RELATED]-(n)
+            WHERE %(active_rel_filter)s AND active_edge.status ="active"
+            CREATE (rl)<-[deleted_edge:IS_RELATED $rel_prop]-(n)
+            SET deleted_edge.hierarchy = active_edge.hierarchy
+            WITH rl, active_edge, n
+            WHERE active_edge.branch = $branch AND active_edge.to IS NULL
+            SET active_edge.to = $at
+            RETURN
+                n.uuid as uuid,
+                n.kind as kind,
+                rl.name as rel_identifier,
+                "inbound" as rel_direction
+        }
+        RETURN DISTINCT uuid, kind, rel_identifier, rel_direction
+        """ % {
+            "active_rel_filter": active_rel_filter,
+        }
+
+        self.add_to_query(query)
+
+    def get_deleted_relationships_changelog(
+        self, node_schema: NodeSchema
+    ) -> list[RelationshipCardinalityOneChangelog | RelationshipCardinalityManyChangelog]:
+        rel_identifier_to_changelog_mapper = {}
+
+        for result in self.get_results():
+            peer_uuid = result.data["uuid"]
+            if peer_uuid == self.node_id:
+                continue
+
+            rel_identifier = result.data["rel_identifier"]
+            kind = result.data["kind"]
+            deleted_rel_schemas = [
+                rel_schema for rel_schema in node_schema.relationships if rel_schema.identifier == rel_identifier
+            ]
+
+            if len(deleted_rel_schemas) == 0:
+                continue  # TODO Unidirectional relationship changelog should be handled.
+
+            if len(deleted_rel_schemas) > 2:
+                raise ValueError(f"Duplicated relationship schema with identifier {rel_identifier}")
+
+            if len(deleted_rel_schemas) == 2:
+                # Hierarchical schema nodes have 2 relationships with `parent_child` identifiers,
+                # which are differentiated by their direction within the database.
+                if rel_identifier != PARENT_CHILD_IDENTIFIER:
+                    raise ValueError(f"Duplicated relationship schema with identifier {rel_identifier}")
```

Comment:
Are there other cases where a schema node could have multiple relationships with identical identifiers? I don't think so but if that's the case this check should be removed